### DISCUSSION TITLE
[atari] binding for Diamond GOS.

### DISCUSSION THREAD

**Discussion by tschak909** [cc65/cc65#1926] [Show and tell]
I have a binding for Diamond GOS, here:
https://github.com/tschak909/diamond-gos-cc65

Seeing as there are GEOS targets, Would you guys like to fold this, into CC65?

-Thom

**Comment by mrdudz** [comment]
What is this "Diamond GOS"? :)

But, generally, yes, sure. You'd have to put it under the same license as the rest of cc65 though, and library code should follow the codestyle outlined in Contributing.md

**Comment by tschak909** [comment]
I just spent the last two days straight making the binding. am very tired.
can you look at it?

As for Diamond GOS, it was an Atari 8-bit cartridge produced by Reeve
Software, containing a full graphical interface, with API. More info here:
https://seriouscomputerist.atariverse.com/pages/dos/dos.other.htm

-Thom

**Comment by mrdudz** [comment]
Just took a brief look, the code looks clean and acceptable to me - however i can not comment on how this can be merged into the project. I guess the gos_xy functions should go into the atari lib, and the others into samples/atari/gos or sth? @groessler ?

**Comment by groessler** [comment]
Hi Thom,


On 11/19/22 22:22, Bob Andrews wrote:
>
> Just took a brief look, the code looks clean and acceptable to me - 
> however i can not comment on how this can be merged into the project. 
> I guess the gos_xy functions should go into the atari lib, and the 
> others into samples/atari/gos or sth? @groessler 
> <https://github.com/groessler> ?
>

I'm not so sure. It looks like it could alternatively be bundled into a 
3rd-party library, unrelated to the official cc65 code base. It does 
need special linker .cfg files, but ok, these could come with the library.

OTOH, there are the GEOS CBM and Apple2 targets. To me they appear to be 
much more invasive to the original targets than the Diamond GOS changes, 
so they probably got their target "blessing" from this.

Thom, would you commit to maintain a "D-GOS Atari" target? The "target" 
part might be just to add some DGOS specific linker cfg files, and the 
code you provide appears to be quite standalone (not changing anything 
of the base Atari target). Another part of maintaining a target will be 
to handle user problems.

regards,
chris

**Comment by tschak909** [comment]
*shrug* ok :)

-Thom

**Comment by polluks** [comment]
By the way do you know https://github.com/ytmytm/geos-atari?

**Comment by tschak909** [comment]
Yes, I am familiar, but I am concerned that given what's left to implement,
whether they will actually do it. There are some formidable challenges to
solve.
-Thom
